### PR TITLE
fix: tsconfig `extends` option

### LINKED ISSUE #82: Error with tsconfig `extends` option set to a node module path
**Bug report**

Having the `extends` option set in tsconfig.json to a path referring to a node module results in an error.

**Input code**

*tsconfig.json*
```json
{
  "extends": "my-module/tsconfig",
  "include": ["./src/**/*"],
  "compilerOptions": {
    "baseUrl": "./src/",
    "outDir": "./dist/"
  }
}
```

```shell
$ dts-bundle-generator -o dist/index.d.ts src/index.ts
```

**Expected output**

```shell
Compiling input files...
Compiler option `outDir` is not supported and will be removed while generating dts
Processing src/index.ts
Writing src/index.ts -> dist/index.d.ts
Checking generated files...
Done in 3.70s
```

**Actual output**

```shell
Compiling input files...
message TS6096: File 'my-module/tsconfig' does not exist.
Error: Error while processing tsconfig compiler options
error Command failed with exit code 1.
```

**Additional context**

After investigation it seems that the `basePath` argument (3rd) to [`ts.parseJsonConfigFileContent`](https://github.com/timocov/dts-bundle-generator/blob/6630db8c1d05cf40eb76e0d0cd5cf0a750c48eae/src/get-compiler-options.ts#L23) has to be an absolute path for the `extends` module resolution to work properly.

I'll submit a pull request in a minute.

### PR DESCRIPTION
**Pull request by medfreeman**
Set ts parseJsonConfigFileContent `basePath` as an absolute path

Fixes #82

### CODE REVIEW

File: src/get-compiler-options.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export function getCompilerOptions(inputFileNames: ReadonlyArray<string>, prefer
 	const configParseResult = ts.readConfigFile(configFileName, ts.sys.readFile);
 	checkDiagnosticsErrors(configParseResult.error !== undefined ? [configParseResult.error] : [], 'Error while processing tsconfig file');
 
-	const compilerOptionsParseResult = ts.parseJsonConfigFileContent(configParseResult.config, parseConfigHost, path.join(configFileName, '..'));
+	const compilerOptionsParseResult = ts.parseJsonConfigFileContent(configParseResult.config, parseConfigHost, path.join(process.cwd(), path.dirname(configFileName)));
```
Comment (timocov) [inline src/get-compiler-options.ts]: What if `configFileName` is an absolute path? Will it works in that case too?

Does `path.join(process.cwd(), path.dirname(configFileName))` return the same (or similar) value what returns tsc's code [here](https://github.com/Microsoft/TypeScript/blob/3f3444be80dc939eb20db52fa98a206dba31e077/src/compiler/commandLineParser.ts#L1328) (`getNormalizedAbsolutePath(getDirectoryPath(configFileName), cwd)`)?

**Comment by timocov** [comment]
Hey @medfreeman! Thanks for your contributing! Please take a look at my comment https://github.com/timocov/dts-bundle-generator/pull/83#discussion_r271501758.

File: src/get-compiler-options.ts (hunk 1 above)
Comment (medfreeman) [reply src/get-compiler-options.ts]: I don't think so.
I'll test and adapt this right now!

Comment (medfreeman) [reply src/get-compiler-options.ts]: done! i'd have preferred to directly import the ts functions, but they're not directly exported, so i used node ones.

**Pull request merged**